SOP0009: Withdraw a dataset

M-casado opened the pull request:
<!--- Thank you for taking the time to contribute to this project! -->
<!--- Provide a general summary of your changes in the Title above -->

## Summary
Added a new SOP0009 with the main skeleton of how to handle a dataset withdrawal request by the VHD Officer, OC/SDPC delegate and Technical Leads (TL).

## Types of changes
<!--- Mention the type of contribution this PR represents: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change which adds new content)
- [x] Modified content (non-breaking change which modifies existing content)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Motivation and Context
Requested SOP: https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/55

### References
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/55

## Changes Introduced
- New document at ``sops\european-level\GDI-SOP0009_dataset-withdrawal.md``
- Minor changes to SOP template.
- New ``.gitignore`` file to ease commits.
- Updated ``sops/README.md`` index table.
- Updated charter document with new terms and definitions.

## Review
Not yet.

## Additional Notes
Something important to note is that **this SOP does not tackle dataset withdrawal from individual GDI Systems**. Instead, it focuses on the workflow and responsibilities without getting to technicalities. This means that **per-system dataset withdrawal SOPs will be required** to fulfill this SOP in a structured manner. Either that, or trust each TL.

## Checklist:
<!--- Review carefully, ticking (or adding an `x` to) each box  upon completion of the item, ensuring you meet the standards required for this PR. -->

**General Compliance:**
- [x] My changes follow the code style of this project ([GDI SOP Style Guide](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_style-guide.md)) and the file naming conventions of the [file accessioning proposal](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs//GDI-SOP_sop-accessioning.md).
- [x] I have verified that all new updated content is accessible, including checking that all external references are readable (i.e., no broken links). These may include references to external resources that should be resolvable, and internal references among SOPs.
- [x] I have properly added this PR's changes to the repository [CHANGELOG.md](../CHANGELOG.md).

Only applicable if the PR includes new, or changes to, GDI SOPs (i.e., documents at ``sops/``):
- [x] My SOP-related changes adhere to the [Generic SOP Template](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_sop-template.md), including format and required fields.
- [x] I have consulted the [Charter](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_charter.md), [ISM](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_information-service-management.md), and [ORR](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_organisational-roles-and-responsibilities.md) documents to ensure compliance. 
- [x] I am complying with the established procedure for SOP creations and modifications, including respecting review phases and notifying needed contributors for reviews.

M-casado committed:
Amend wrong index numbers of procedure

M-casado committed:
SOP0009 - Initial draft

M-casado committed:
SOP0009 - Reorganize glossary

M-casado committed:
SOP0009 - Small modifications to text

M-casado committed:
SOP0009 - Add diagram

M-casado committed:
SOP0009 - Update charter glossary

M-casado committed:
SOP0009 - Update glossary

M-casado commented:
The linter is failing for two reasons:
- I'm missing reviewers and approvers -- @GabiRinck, appoint some if possible.
<img width="1217" height="118" alt="image" src="https://github.com/user-attachments/assets/058ca421-2d30-4eb3-8418-76191b0230ce" />

- Some URLs in the email templates I have are "for the future", once the SOP is actually in place in the ``main`` branch. So these will not be resolved. This means that evaluation of the workflows need a finer detail to assess if they're correct or not.
<img width="1259" height="266" alt="image" src="https://github.com/user-attachments/assets/a529139e-1412-46ba-9459-478ac9b96574" />

M-casado committed:
Add .gitignore

M-casado committed:
add missing tabulate dependency

M-casado committed:
SOP0009 - Amend dates and AF acronym

M-casado committed:
SOP0009 - Add new line to SOP index table

M-casado committed:
Update CHANGELOG

M-casado committed:
SOP0009 - Update scope

M-casado committed:
SOP0009 - Minimal diagram change

M-casado committed:
SOP0009 - Add links to requested SOPs

M-casado commented:
See relevant [**slack thread**](https://gdi-elixir.slack.com/archives/C06RJQJNHFC/p1752606288102789), with discussions over soft-deletion, EDIC, central storage and overall process.

M-casado commented:
@silviabah & @kjellp volunteered as reviewers - @GabiRinck volunteered as approver

M-casado commented:
Reminder for **reviewers** about the handy review checklist: https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_review-checklist.md

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+| Authorizer |                 |                 |              |
+
+### 4. Purpose
+The purpose of this SOP is to define the process for withdrawing a dataset when requested by its controller European Genomic Data Infrastructure (GDI). Following its procedure ensures the removal is propagated swiftly and consistently across all services while meeting legal, ethical, and audit requirements. With it, GDI nodes protect data-subject rights, prevent continued exposure of withdrawn da ...
```
```suggestion
The purpose of this SOP is to define the process for withdrawing a dataset when requested by its controller from the European Genomic Data Infrastructure (GDI). This procedure ensures the removal is propagated swiftly and consistently across all services while meeting legal, ethical, and audit requirements. With it, GDI nodes protect data-subject rights, prevent continued exposure of withdrawn data, and maintain a transparent, documented record of the action.
```

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+The purpose of this SOP is to define the process for withdrawing a dataset when requested by its controller European Genomic Data Infrastructure (GDI). Following its procedure ensures the removal is propagated swiftly and consistently across all services while meeting legal, ethical, and audit requirements. With it, GDI nodes protect data-subject rights, prevent continued exposure of withdrawn da ...
+
+### 5. Scope
+This procedure applies to **all datasets within the GDI**, regardless of which system or node they were originally submitted to. It covers datasets submitted via any GDI component, including but not limited to: **Federated EGA** (FEGA), the **GDI Beacon network**, the **GDI User Portal catalog**, and **FAIR Data Portals** (FDP) under GDI (see more details at [Step 7](#87-per-system-dataset-withdr ...
```
I think withdrawal from FEGA should be out of the scope of this SOP. As the request for deletion is outside the GDI infrastructure

M-casado replied:
At the time I wrote it, I think FEGA was going to be the model for GDI, and it was a confusing period. As of today, I think I completely agree with you.

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+
+As a European-level SOP, it is designed to be directly implementable by all GDI nodes and services. The SOP addresses both **full dataset withdrawals** (removing an entire dataset from GDI) and **partial withdrawals** (removing or retracting a portion of a dataset, such as specific individuals' data). This SOP focuses on the **overarching process and coordination** required to withdraw the data a ...
+
+It encompasses **withdrawals triggered by a requester**. This entity may be the data controller, a data subject (e.g., withdrawing their consent), legal or ethical entity fulfilling their obligations, or any GDI member due to dataset quality issues or technical needs.
```
The request should just come from the legal entity that signed the DPA with the GDI node. i.e data subject should contact the data submitter, and then the data submitter requests the withdrawal to the GDI node. Same as legal or ethical entities, they should communicate first with the data controllers and then the data controllers make the request to GDI.

M-casado replied:
Happy to rephrase. Out of curiosity, how are requests handled if not from the Data controller, and the data controller is unresponsive?

As far as I understand, GDPR covers personal data subjects even when the data controller is unresponsive.

M-casado replied:
I think that in this SOP I included that "first communicate with the data controller" step, as part of the HD steps for that reason. Should I remove that from GDI's plate? Would the response to the data requester be simply "get the data submitter to request the withdrawal to us"?

M-casado replied:
Fixed

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+_Out of Scope_: This SOP does **not** cover the initial amendments during submission of datasets or routine data updates. It also does **not** cover suspension of _access_ without removal. Any system-specific technical steps for deletion (e.g., how to remove metadata from a GDI Beacon) are referenced but detailed in separate SOPs for those systems (see [Step 7](#87-per-system-dataset-withdrawal)) ...
+
+### 6. Introduction and Background Information
+The GDI is a federated network of national and European services that together enable discovery, access, and analysis of genomics and related health data. Because datasets may be copied, indexed, or exposed by multiple GDI components (e.g., Federated EGA, Beacon, FAIR Data Portals), withdrawing a dataset requires a coordinated cross-system process. This SOP supplies that process.
```
FEGA is out of the scope of the GDI infrastructure? i.e the withdrawal should then be requested to FEGA and I would guess since the data is not longer available on FEGA, GDI services won't be able to retrieve.

M-casado replied:
In the scenario where a GDI node and FEGA node overlap data withdrawal should happen simultaneously on "both levels". If they don't overlap, there may be a dataset withdrawal request to GDI, but that should effectively propagate in some way to the FEGA node/storage hosting the data.

On the other hand, removing it from GDI features would "cover" the GDI SOP scope, so your point of view is also valid.

M-casado replied:
I'm happy to just remove that part and let each node handle the services they withdraw from independently.

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+| ``1`` | Anytime a person or organisation submits a ``dataset-withdrawal-request`` to the GDI Virtual Helpdesk (VHD) | VHD officer |
+
+Once the request is received, it is your responsibility as VHD officer to evaluate its idoneity:
+- The request **must contain**:
```
The first thing to check is that the "requester" has the right authority to request the withdrawal. So this would be the "authorised submitter" ot the entity that has signed legal documents with the GDI node. 
As well, I think this request should be done at the node level, since the node is hosting the data files and the metadata and from there is retrieved to the GDI services? So potentially, if the metadata/files are no longer available from the node the GDI services won't be able to retrieve. 

Thus, I would suggest the VHD refers the user to the Node HD to follow up on the data withdrawn.

M-casado replied:
> The first thing to check is that the "requester" has the right authority to request the withdrawal

✔️ 

> I think this request should be done at the node level

Albeit I agree that ultimately it's done at the node level, I think this SOP was intended to be used GDI-wide. I would expect a withdrawal request may not always land on the "node" that actually hosts all data to be withdrawn, right? Therefore, I think it's the responsibility of GDI as a whole to redistribute the request appropriately to the nodes/services that need to get rid of it.

> I would suggest the VHD refers the user to the Node HD to follow up on the data withdrawn

So far in this SOP I haven't made the distinction between VHD and node HD, since both may end up being the same people and it's not clear to me still what the structure will end up being like in reality.

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+
+**Only when** all '✔️' requirements are met, can the process can move forward:
+- If requirements (✔️) are met, move to ⏩[**step 4**](#84-log-and-acknowledge-valid-request)
+- If requirements (✔️) are **not** met, the request is **flagged as invalid**. Move to ⏩[**step 2**](#82-communicate-rejection-to-requester).
```
This should be the first step to check when receiving the request. Not sure if all this information will be available at European level or the Node Level. i.e, Who is the Data controller?
I would guess this will be at a Node level so this request seems to be more suitable at a node level.

M-casado replied:
I think in reality **it is** almost like the first step. The step (``8.1``) before is simply confirming that the request follows the expected format/content. I think that needs to happen before you actually check the authority of the request: you wouldn't want to waste time checking the authority of the person when the dataset ID is missing, or when the dataset they refer to hasn't been released, right?

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+| ``4``             |After requester authority is verified|VHD Officer|
+
+Given that the request is valid and the requester has authority to withdraw the dataset:
+- **Create a VHD ticket** to tackle the dataset withdrawal itself:
```
Operations ticket, i.e JIRA board as this can be confused with a VHD ticket i.e RT queue

M-casado replied:
Correct me if I'm wrong, but I think so far the ticket management system of the VHD is not clear, and it would be a queue system.

I don't know if this will include both queue of user responses and operational tasks, or just one or the other. I'm not sure myself what the best way to write it would be.

M-casado replied:
So the way I originally wrote this part I think it was meant to say "whatever management system you have, there should be a new item in the queue, for you to actually follow through" 🤔

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+Given that the request is valid and the requester has authority to withdraw the dataset:
+- **Create a VHD ticket** to tackle the dataset withdrawal itself:
+   - Create it with **type** ``dataset-withdrawal``.
+   - **Link it** to the initial ``dataset-withdrawal-request`` VHD ticket that was generated with the requester communication. As per your system, this link should imply that the ``dataset-withdrawal-request`` ticket "is blocked by" this new ``dataset-withdrawal`` ticket.
```
The URL of the ticket with the request has to be "referred" in the Operations ticket. Probably, this will be on different platforms.

M-casado replied:
Similar to my comment above, since we don't know yet, and the SOP0008 from Elisavet had different categories for tickets that **include** things like these... I don't know if to separate it or leave it together for now.

I think it would be best to leave it generic, to let each GDI node adapt it to their ticketing system (?)

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+2. Update the VHD ticket with assigned leads and expected completion timeline.
+
+See more details at [Step 5](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/sops/european-level/GDI-SOP0009_dataset-withdrawal.md#85-assign-withdrawal-response-team).
+
```
This approach may be too complex and time-consuming, especially considering the volume of data coming from multiple Nodes. Since GDI services are potentially synchronising continuously with all Nodes, the key should be that when a dataset is withdrawn at the Node level, the Node sends a technical withdrawal message (e.g., via API communication or RabbitMQ). This message should update the dataset status from "available" to "withdrawn".
Once this status is updated, GDI services should either stop synchronising the dataset or remove the corresponding record so it is no longer visible in any service.
Therefore, the focus should be on how the Node communicates a withdrawal event—that is, how the Node sends the technical "withdrawn" message so GDI services immediately stop exposing the data. This process should be fully automatic: when a Node marks its data as "withdrawn", the change is propagated to the GDI services.
Then the Helpdesk officer just needs to verify that, after the withdrawal at the Node level, all GDI services have synchronised correctly and the dataset is no longer available.

M-casado replied:
I think we're referring to different things. With ``GDI systems where this dataset (or the specified subset) resides`` I meant, for example, the Frequency Allele Beacon of the GDI node that hosts the data; or the storage where the actual datafiles are hosted in a GDI Beacon; or the metadata from a node FAIR data point.

It's not that "every consumer point of GDI" (i.e., a component that doesn't have the data, but instead reads it from the GDI Node) will need to be identified, but instead that every component where the data actually **resides** needs to be identified. This has to happen before what you comment on: propagation of withdrawal to the consumer components.

I completely agree that the event needs to be propagated automatically, but the "hosting" of the data needs to be identified in a manual way, I think: as of now, data in GDI may be in a beacon, allele frequency beacon, a FEGA storage, etc.

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
`````diff
@@ -0,0 +1,466 @@
+GDI Virtual Helpdesk
+````
+
+- **Send acknowledgement to requester** within 5 working days. Use the following template, including ticket ID and summary of next steps.
`````
As the ticket ID will be an internal "operations" ticket, this information does not need to be shared with the user. As well, the answer should be sent in the original request ticket the user created to request the withdrawal since this is a follow-up.

M-casado replied:
I was referring to the internal ticket ID that was created (I imagine) automatically at reception of the request. The one that commonly RT tickets add to the subject for traceability.

I'll reword the phrase 👍

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
`````diff
@@ -0,0 +1,466 @@
+````
+- Once a TL has been assigned (i.e., each has acknowledged it) to each of the GDI systems, **record assignments and target completion date** in the ``dataset-withdrawal`` ticket.
+- **Notify VHD officer** (who managed the initial request) that leads are assigned. Include in the communication _who_ has been assigned _what_.
+- This **concludes** your responsibilities as the OC/SDPC delegate with respect to this SOP. 🔚
`````
As mentioned above, this should be an automatic process when the node withdraws the data.

M-casado replied:
Similar to my comment above, this is referring specifically to that "node withdraws the data" part. The TLs are tasked to "withdraw" the data from the systems that the GDI node uses to host the data (e.g., Beacon).

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+While the ``dataset-withdrawal`` ticket is being solved (see [step 5](#85-assign-withdrawal-response-team) and [step 7](#87-per-system-dataset-withdrawal)), your role as VHD Officer is to:
+- **Track status of ``dataset-withdrawal`` ticket** (including subtasks). This includes asserting that TLs are following up on their duties with regards to their GDI systems, and reminding them if not.
+- **Follow up on overdue tasks** and escalate to OC/SDPC if needed.
+- **Facilitate communication** between requester, TLs and committees.
```
The communication should be kept a minimal i.e, requesters should not interact with TLc and committees. Requesters should just communicate with the HD officer.

silviabah commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -0,0 +1,466 @@
+| Step identifier   | When| Who |
+|:------------------|:----|:----|
+| ``7``             |After appointed as TL by the OC/SDPC at [Step 6](#86-track-dataset-withdrawal)|Technical Lead (TL)|
+
+As the TL of a GDI system, your tasks are:
+1. **Identify dataset presence**. Determine whether any (meta)data for the dataset to be withdrawn resides within the specified GDI system. This includes any type of **submitted** (meta)data, from raw data files to metadata about a patient.
+2. **Remove dataset's (meta)data**.  If present, execute withdrawal in line with the relevant system-specific SOP and the requested scope (e.g., partial/full withdrawal, soft/hard deletion).
+3. **Report status**. Update relevant stakeholders (mainly VHD officer in charge of the ticket) about your progress within this step.
+
+To accomplish these tasks, **follow the appropriate SOP** for your specified system:
+- **LS Login** (_#! SOP TBD_):
+   - Nodes' Identity Provider (IdP) (_#! SOP TBD_).
+   - Node Access Management (_#! SOP TBD_).
+- **User Portal** (UP) (_#! SOP TBD_).
+   - Resource Entitlement Management System (REMS) (_#! SOP TBD_).
+   - Beacon Browser (_#! SOP TBD_).
+      - Nodes' Beacons (_#! SOP TBD, see https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/64_).
+   - Dataset Browser (_#! SOP TBD_).
+      - Nodes' FAIR Data Points (FDP) (_#! SOP TBD, see https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/63_).
+   - Allele Frequency Browser (_#! SOP TBD_).
+      - Nodes' Allele Frequency Beacons (_#! SOP TBD, see https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/65_).
+- **Nodes' Secure Processing Environment** (SPE) (_#! SOP TBD_).
+
+Once you have completed the relevant withdrawal SOP for the specified GDI System:
```
This is a good checklist that the Node HD can check after doing the withdrawal, and just then, if the action is not being propagated to the GDI services, first check internally with their dev team that the communication with the services is working correctly, and if the devs confirm,, then escalate the issue to the TL of the services.

M-casado replied:
Isn't it a bit too technical for the HD officer to do? It covers so many possible components of GDI that I would be surprised if someone knew all of these. At least, that is, until SOPs to check each were done.

silviabah reviewed (commented):
Hello, I added a few comments about the workflow of the SOP :)

M-casado committed:
Update sops/european-level/GDI-SOP0009_dataset-withdrawal.md

Co-authored-by: silviabah <80835063+silviabah@users.noreply.github.com>

M-casado committed:
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62#discussion_r2565102090

M-casado committed:
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62#discussion_r2565500658

M-casado committed:
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62#discussion_r2565572727

M-casado committed:
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62#discussion_r2565588358

M-casado committed:
Add reviewers/Approvers

M-casado committed:
Address feedback from Kjell

M-casado committed:
Update last edit date

M-casado committed:
Minor fixes

M-casado commented:
Feedback from Kjell was addressed; same for most comments from Silvia, a few still to discuss.

Since remaining changes are relatively small, you may want to review for approval already @GabiRinck

M-casado committed:
Add skippable uppercase list to script

M-casado committed:
Addressing Silvia's feedback

M-casado commented:
Based on your review, @silviabah, I added more changes here: https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62/commits/9374c5da5b0663db6282f37d5468ea7197cf2658

Feel free to review further, but I think I added all requested changes we discussed.

M-casado commented:
Ticket is ready for approval, @GabiRinck

M-casado committed:
Update glossaries

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -33,6 +33,7 @@ Find GDI SOPs common Glossary at the [**charter document**](https://github.com/G
 | DPO | Data Protection Officer |
 | EBI | European Bioinformatics Institute |
+| EDIC | European Digital Infrastructure Consortium |
 | EGA | European Genome-phenome Archive |
```
delete row - on EGA mentioned in the document

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -69,24 +71,24 @@ See qualifications and responsibilities of the roles at the [**Organisational Ro
 
 ### 5. Scope
-This procedure applies to **all datasets within the GDI**, regardless of which system or node they were originally submitted to. It covers datasets submitted via any GDI component, including but not limited to: the **GDI Beacon network**, the **GDI User Portal catalog**, and **FAIR Data Portals** (FDP) under GDI (see more details at [Step 7](#87-per-system-dataset-withdrawal)). 
+This procedure applies to **all datasets within the GDI**, regardless of which system or node they were originally submitted to. It covers datasets submitted via any GDI component (i.e., storage and interfaces), including but not limited to: the **GDI Beacon network**, the **GDI User Portal catalog**, and **FAIR Data Portals** (FDP) under GDI (see more details at [Step 7](#87-per-system-dataset-w ...
```
As far as I know, FDP stands for "FAIR Data Point", not "FAIR Data Portals"

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -69,24 +71,24 @@ See qualifications and responsibilities of the roles at the [**Organisational Ro
 
 ### 4. Purpose
-The purpose of this SOP is to define the process for withdrawing a dataset when requested by its controller from the European Genomic Data Infrastructure (GDI). This procedure ensures the removal is propagated swiftly and consistently across all services while meeting legal, ethical, and audit requirements. With it, GDI nodes protect data-subject rights, prevent continued exposure of withdrawn da ...
+The purpose of this SOP is to define the process for withdrawing a dataset from the European Genomic Data Infrastructure (GDI) when requested by its data controller. This procedure ensures the removal is propagated swiftly and consistently across all services while meeting legal, ethical, and audit requirements. With it, GDI nodes protect data-subject rights, prevent continued exposure of withdra ...
```
In this paragraph you say "requested by its data controller" vs in the next section "Scope" you list a number of entities/roles who could request withdrawal. Perhaps go with a generic wording here like "upon request" instead of "requested by its data controller"

M-casado replied:
You may have clicked on an old commit when doing this comment (?) - There's an "Outdated" flag in it, and when I checked the wording, it's different in the most [up-to-date version of the file](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/issues/55/sops/european-level/GDI-SOP0009_dataset-withdrawal.md#4-purpose) in my branch (the one I'm trying to merge with the main project).

M-casado replied:
The current phrasing is more generic indeed:

> Requests can be initiated by many parties, but execution requires controller authorisation or a valid legal basis and governance decision

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -69,24 +71,24 @@ See qualifications and responsibilities of the roles at the [**Organisational Ro
 As a European-level SOP, it is designed to be directly implementable by all GDI nodes and services. The SOP addresses both **full dataset withdrawals** (removing an entire dataset from GDI) and **partial withdrawals** (removing or retracting a portion of a dataset, such as specific individuals' data). This SOP focuses on the **overarching process and coordination** required to withdraw the data a ...
 
-It encompasses **withdrawals triggered by a requester**. This entity may be the data controller, a data subject (e.g., withdrawing their consent), legal or ethical entity fulfilling their obligations, or any GDI member due to dataset quality issues or technical needs.
+It encompasses **withdrawals triggered by a requester**. This person or entity may be the data controller, a data subject (e.g., withdrawing their consent), legal or ethical entity fulfilling their obligations, or any GDI member due to dataset quality issues or technical needs. Examples of these include: (1) data subjects requesting withdrawal based on their rights; and (2) controllers needing to ...
```
"legal or ethical entities" - as we've just discussed, I'm not sure about the wording here. Legal entities ok, but this could be anybody (a bit too wide), not sure what an ethical entity would be... "authority" instead of "entity" would perhaps make more sense in this context, but, as discussed, was probably not what you wanted to say. I would suggest that we just delete ", legal or ethical entity fulfilling their obligation" to keep it simple & avoid ambiguity. 
Also, how can GDI accept a withdrawal request from a data subject directly if the data are pseudonymised??

M-casado replied:
It seems like, as above, your comment is referring to an old version of the file. The latest one says:

> It encompasses **data withdrawals triggered by a requester** (including data subjects). Note that execution must be authorised by the relevant data controller, or otherwise have a valid legal basis and governance decision. Examples of these include: (1) data subjects requesting withdrawal based on their rights; and (2) controllers needing to withdraw data (without an explicit user requesting it) to comply with regulations.

This way, I think your first comment about _ethical entity_ is addressed.
Regarding your second comment, my grain of salt is:

- Data can be pseudonymised, but still be personal data. A data subject can still request to withdraw their data from a GDI dataset.
- Even if not considered personal data (as per the latest Court of Justice resolution, which gave new insights to the GDPR definitions) "to the general user", it will still be personal data "for GDI". To that extent, a data subject can still request their removal from a dataset.

GabiRinck replied:
I do agree that a data subject has a right to have their data deleted, but I honestly can't see how they can come to GDI / the node & request deletion of their data directly if all the data a pseudonymised. I would have to come through the submitter of the data who holds the key & identify exactly which record(s) to delete and which not. (Once they set up some sort of syste for dynamic consent which then links the login credentials of a particular data subject to one specific record somehow, then this would be possible to request this directly, but for the time being, I just can't imagine how the data subject can request deletion from a node. I think they would have to go through the submitter (or whoever collected the consent??)
How do you think this would work in practice??

GabiRinck replied:
Also in 8.3 you say "The request must come from the corresponding dataset controller. Therefore, if the requester does not fit that role, flag the request as invalid..." 
The data subject is usually not the data controller - so then you're saying here in 8.3 that the data subject can not make a valid request.... hence I would not confuse things & mention it in the text above!??

M-casado replied:
> I honestly can't see how they can come to GDI / the node & request deletion of their data directly if all the data a pseudonymised

Trying to put myself in their shoes, I would say I could have given my data to project X of the NHS, and then I see that GDI has datasets from that project. Even if the data is pseudonymised, my personal data is contributing in some way to those datasets. I could then request the withdrawal of my data to GDI. If that happened, as per the SOP, we would refer them to the data controller, which would ultimately be the one requesting the partial withdrawal.

> identify exactly which record(s) to delete and which not

In my opinion it can be more broad. A data subject may say "Whatever my personal data is contributing to this dataset, I want it removed". Whether it's a valid request or not, that's a different bit.

I still see the value of having the "data subjects" named in the SOP, as the support officer going through it may find it helpful: "I got a request by a data subject and... ah, alright, refer to data controller".

> I think they would have to go through the submitter (or whoever collected the consent??)

We could reword it a bit, but I think precisely this is what the SOP says. For example, in:

- [Scope](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/issues/55/sops/european-level/GDI-SOP0009_dataset-withdrawal.md#5-scope), we say that the SOP encompasses dataset withdrawal requests made by requesters, including data subjects. That doesn't mean that all requests are valid from all requesters, but this SOP includes how to handle the invalid ones as well. 
-  [Step 8.2](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/issues/55/sops/european-level/GDI-SOP0009_dataset-withdrawal.md#82-communicate-rejection-to-requester), we say that we request more information to the requester, especially if they are data subjects that skipped the data controller.

Shall we reword the "Examples of these include..." from the Scope paragraph?

M-casado replied:
I just changed the examples. It should read better now.

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -69,24 +71,24 @@ See qualifications and responsibilities of the roles at the [**Organisational Ro
 
 ### 6. Introduction and Background Information
-The GDI is a federated network of national and European services that together enable discovery, access, and analysis of genomics and related health data. Because datasets may be copied, indexed, or exposed by multiple GDI components (e.g., Federated EGA, Beacon, FAIR Data Portals), withdrawing a dataset requires a coordinated cross-system process. This SOP supplies that process.
+The GDI is a federated network of national and European services that together enable discovery, access, and analysis of genomics and related health data. Because datasets may be copied, indexed, or exposed by multiple GDI components (e.g., GDI Storage, Beacon, FAIR Data Portals), withdrawing a dataset requires a coordinated cross-system process. This SOP supplies that process.
```
"FAIR Data Point" instead of "FAIR Data Portals" or just "FDP"

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
"idoneity" - Nice word I've never seen before, but as there will be nodes all across Europe & many GDI members will not have English as a native language, I would try to avoid unusual words like this. I would just stick to validity, the same word you used in the flow chart.

M-casado replied:
It is very common in spanish language 🥲

M-casado replied:
I'll swap it

M-casado replied:
Funnily enough, it was indeed another obsolete comment. It is ``validity`` in the current version of the file.

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
What about including the information that the ticket will be closed if no response is received within 15 days??

M-casado replied:
I'm no expert on what to tell users and what not to tell. This information (the SOP) will be publicly available (although I doubt a user goes all the way to the SOP to find the response timings).

I asked EGA HD and normally we just add a trailing message such as the following, not to bound ourselves with the timings:

> Please let me know if you have any questions,
I will now close this ticket, but if you reply with an update or report that the issue persists, it will be reopened, and we will continue to assist you.
Kind regards,

GabiRinck replied:
GDI I have not decided on a VHD ticketing system, so we don't know if a ticket can be reopened after it has been closed... I honestly think it's not a realistic expectation that the user would check the GDI SOPs and this shouldn't be expected of them!! It's all about managing expectations and being transparent helps. If you do set a deadline for closing the ticket after a certain number of days, then I don't know why you would object to mentioning this in the response to the user, too. No reason to be secretive about this!?? We have nothing to hide!??

M-casado replied:
> I honestly think it's not a realistic expectation that the user would check the GDI SOPs and this shouldn't be expected of them

That's precisely what I meant: I **don't expect** users to do that, but neither had in mind that the timings were to be cited in the email to the users. Simply put, we would have a rather "open to help" ending by the end of the email, similar to EGA's, and not tie ourselves with timings. The reason being that the more we say in the email, the more we can shoot ourselves as maintainers in the foot when human mistakes happen (e.g., a HD officer closing the ticket after 13 days instead of 15).

M-casado replied:
Given that it's not a critical bit of the SOP, I don't mind adding it. I'll reword this bit to include that information.

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
```diff
@@ -251,15 +253,15 @@ With the requester's details and available resources:
-1. Has the **dataset submitter** been made aware of the request?
-   - If so, have they have their right to **challenge** the decision? The ultimate decision is made by the data controller, but data submitters may help finding out possible issues in the request (scope, authority, etc.).
-1. Has **GDI coordination** been made aware?
+   - Is it a **data subject**? If so, does its participant/subject ID match the dataset? In other words, is the participant part of the requested dataset? Since this cannot be determined at the european level, it involves **communication with the Node Contact Point (NCP) and/or the dataset controller** for you to know if the individual is included. If uncertain, you shall request the requester to ...
```
To be honest, I can't see how the data subject can directly request withdrawal from the VHD, they are unlikely to have a subject ID etc?? I would think that they would always have to go through the legal entity whom they have given consent to??  But since you've mentioned it above & Kjell/Silvia were fine with it, I don' t mind if you leave it in. It just seems very odd to me (well once they implemented some tool for dynamic consent, it might work somehow!???)
Also, how do you check that the ID actually  matches this particular person and you don't delete someone else's data if they gave you the wrong ID??

M-casado replied:
Given that this is an outdated comment as well, I would refer to the [latest **version**](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/issues/55/sops/european-level/GDI-SOP0009_dataset-withdrawal.md#83-verify-request-authority). I think steps 8.2 and 8.3 would be in line with your comment.

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
"type" - is this part of the ticket classification? (if it relates to the ticket classification SOP, then I would add a link/reference)

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
as before in step 8.4, I would include "Attach to the email any supporting documentation (e.g., verification of authority, screenshots, initial request...)." So that the TLs can (must!!?) check the ID against the original request to avoid accidentally deleting the wrong data. Perhaps even checking by a 2nd person to be on the safe side.

M-casado replied:
Why would the TLs need to check the ID against the original request? As per the latest version, the request must have the green light by the data controller. A TL would not need to, nor should, do any ID verification. They just need to know the dataset and to what extent the dataset needs to be withdrawn (partial/full).

The only exception where a TL would do any sort of ID handling, is where the dataset withdrawal is _partial_ **and** the extent to withdraw is a single data subject, **and** the way to identify what part of the dataset belongs to that subject is through their ID.

GabiRinck replied:
The response I left in the comment below, is also providing an explanation why I was making this comment, so please see below - perhaps it would be good if we had a quick chat about this so that I can explain my thinking behind this. Generally, as the SOP is about withdrawing/"deleting" data, we need to think about the risks within this process (here I see a risk of potentially deleting data for the wrong dataset or subject if there is an error/typo when passing on information), then we need to look at which steps may have room for (human) error & how we can help to make the step less error-prone, lowering the risk of something going wrong. If you don't forward the information in which the ID was first mentioned (prior to it being transfered to a different message).

GabiRinck commented on sops/european-level/GDI-SOP0009_dataset-withdrawal.md:
I would include some checks of the ID(s) against the forwarded original request, just to minimise the risk of deleting the wrong data.

M-casado replied:
As per my comment above, I think the TLs shouldn't need to do any verification. The only information they need is:
- That a valid (i.e., data controller-approved) request was created.
- What dataset is to be withdrawn
- What the scope of the withdrawal is

GabiRinck replied:
As long as there is a manual step involved in copy & pasting IDs (or god forbid, even typing an ID) into an email, the TL needs to have access to the initial information ensuring that he/she is deleting the correct record. If the transfer of the information was automated with no element/step where one could introduce human error by pasting the wrong ID, then I agree that the TL does not require any original copies of anything. But since we are talking about deleting data (and then passing on the request by sending emails) we need to be extra careful & provide the safety net that the person carrying out the deletion can double-check that it is the correct ID. This is the least we can do....

M-casado committed:
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62#discussion_r2891451676

M-casado committed:
https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62#discussion_r2891451676

M-casado committed:
Address Gabriele's review

M-casado committed:
Amend relative reference

M-casado committed:
Change examples

M-casado committed:
Add deadline information to email to requester

M-casado committed:
Update roles and added risk disclaimer in communication to TLs and Node HD

GabiRinck reviewed (approved):
As discussed, happy to approve. 
Many thanks, great work!
